Gem/Code/Include/UiRoundsLifecycleBus.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
```

Comment:
It's a looot of code for a simple event. Did you consider just having an AZ::Event on the NetworkMatchComponent instead?

Reply:
Ah, like turn NetworkMatchComponent into an AZ::Interface so I can access it from anywhere and subscribe to its _RoundRestTimeRemainingAddEvent_?

Reply:
There's a chance the client's UI will get spawned in before the NetworkMatchComponent has activated itself on the client; trying to think of a nice way to know when NetworkMatchComponent is available. These components are on different entities, so can't leverage component dependency system.

Reply:
Updated! I removing code and are tapping directly into an AZ::Event on NetworkMatchComponent. It's a bummer I have to wait using TickBus to grab NetworkMatchComponent though.

Reply:
Oh, woah.. naw I wasn't thinking to turn the netmatchcomponent into an interface.. UiRestBetweenRoundsComponent and NetworkMatchComponents are both level components on the level entity, right? If so you can just call FindComponent<> on the UiRestBetweenRoundsComponent parent entity and it should give you back the NetworkMatchComponent directly. If you set the component dependencies up right, netmatchcomponent will even be guaranteed to activate first so you can just FindComp<>()->GetEventHandler().Attach() or whatever all in a single line.

Reply:
Since this is a Ui component inside of an LyShine canvas there  components are on different entities, so can't leverage component dependency system.
I’m fine with leaving NetMatchComponent an interface since there will definitely be others that need to access it (example: the player component will need to stop moving in between rounds).
The only cringe is having to wait a tick to ensure the other entity is active, but this is a common pattern in O3D as there isn’t any behavior context set up for knowing when all the entities in a level have been activated.
In the plus side, there’s no mental logic hoping using e-bus, and are tapping into NetMatch directly.

Reply:
Regarding getting to the parent of UI entity. UiRestBetweenRoundsComponent is a canvas UI component. There is a way to get the entity that owns it but it takes an EBus call, and instead you are doing the whole AZ::Entity to avoid EBus calls, it would defeat the whole purpose.

Reply:
It wasn't necessarily to avoid the bus calls as I don't think those will be a performance concern here under any circumstance. It was more to remove the need for the extra hundred lines or so of setup code every bus takes + avoiding any hit on compile times + the simpler call chains if we ever need to step-debug through this stuff.
That said, if the components are on different entities then I doubt a direct entity access approach will end up any cleaner, and the interface Gene is using here is probably the best compromise.

Reply:
Thanks for the feedback and I agree :)
Merging in 3, 2, 1...